client/src/redux/hooks.js

```diff
@@ -0,0 +1,54 @@
+import { useCallback } from 'react';
+import { useDispatch, shallowEqual, useSelector } from 'react-redux';
+import { loadUser, register, login, logout } from './actions/auth';
+
+export function useAuth() {
```

Comment:
Yes absolutely, it could be used in the any components whenever you need a state or actions instead of calling useSelector and useDispatch so much time.